MVC/view/austinguitest.py

```diff
@@ -170,22 +170,21 @@ def backspace(self):
     '''
     def makeGuess(self, *args):
         input = self.e.get()
-        #made it so userGuess returns true/false so that way we only insert valid words into the listbox.
-        if self.controller.checkInput(input.lower(), self.reqLetter.lower()) == True:
+        # Check if the input is within the required length range
+        if len(input) < 4 or len(input) > 15:
+            messagebox.showinfo("Invalid guess", "Word must be between 4 and 15 letters.")
+        # Check if the word has already been guessed correctly
+        elif input.lower() in self.listBox.get(0, tk.END):
+            messagebox.showinfo("Already Guessed", "This word has already been guessed correctly.")
+        # Check if the input is valid
+        elif self.controller.checkInput(input.lower(), self.reqLetter.lower()) == True:
+            # Check if the guess is correct
             if self.controller.controllerUserGuess(input) == True:
                 self.listBox.insert(tk.END,input.lower())
             else:
-                messagebox.showinfo("Invalid Guess", "Please re-enter a guess.")
+                messagebox.showinfo("Invalid Guess", "Word is not in list.")
         else:
-            messagebox.showinfo("Invalid input", "Ensure each guess uses the required letter, and consists of letters only.")
-        #update the points and rank after every guess.
```

Comment:
Why did you remove this part? The makeGuess function needs to still update points, rank, and clear the input box.